pkg/validator/shoot_validator.go

```diff
@@ -126,9 +120,41 @@ func (v *Shoot) validateShootUpdate(ctx context.Context, oldShoot, shoot *core.S
 		}
 	}
 
+	// Only validate against cloud profile when zones are updated.
+	// This ensures that already running shoots won't break after a zone is removed from a cloud profile.
+	if len(infraConfig.Networks.Zones) > len(oldInfraConfig.Networks.Zones) {
```

Comment:
What about the case when one zone is deleted & one other zone is added -> would that not lead to no validation of the infra config against the CloudProfile?

Reply:
Good question!
This immutability is already checked here https://github.com/gardener/gardener-extension-provider-aws/blob/4633575f4689330401f4e0504d306fa7e37bbe94/pkg/apis/aws/validation/infrastructure.go#L153